src/shell/lisa_gen_buildroot_rootfs

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+set -eu
+
+BUILDROOT_URI=git://git.busybox.net/buildroot
+BUILDROOT_VERSION=2018.08
+BUILDROOT_DIR=$LISA_HOME/src/buildroot
+BUILDROOT_PATCHES_DIR=$(dirname $0)/buildroot_patches
+BUILDROOT_CONFIG=$(dirname $0)/buildroot_config
+
+function print_usage {
+	echo "Usage: $0 [options]"
+	echo "	options:"
+	echo "		-p: purge buildroot to force a fresh build"
+	echo "		-h: print this help message"
+}
+
+while getopts "hp" opt
+do
+	case $opt in
+		h)
+			print_usage
+			exit 0
+			;;
+		p)
+			rm -rf $BUILDROOT_DIR
+			exit 0
+			;;
+		?)
+			print_usage
+			exit -1
+			;;
+	esac
+done
+
+# Executed command @$1 if file @$2 doesn't exist
+function do_if {
+	if [ ! -e $2 ]; then
+		$1
+		touch $2
+	fi
+}
+
+function br_clone {
+	git clone $BUILDROOT_URI $BUILDROOT_DIR
+}
+
+function br_checkout_version {
+	pushd $BUILDROOT_DIR > /dev/null
+	git checkout -b $BUILDROOT_VERSION $BUILDROOT_VERSION
+	popd > /dev/null
+}
+
+function br_patch {
+	pushd $BUILDROOT_DIR > /dev/null
+	for patch in $(ls $BUILDROOT_PATCHES_DIR)
+	do
+		git am $patch
+	done
+	popd > /dev/null
+}
+
+function br_apply_config {
+	cp $BUILDROOT_CONFIG $BUILDROOT_DIR/.config
+	make -C $BUILDROOT_DIR olddefconfig
+}
+
+
+do_if br_clone $BUILDROOT_DIR/.lisa_cloned
+
+do_if br_checkout_version $BUILDROOT_DIR/.lisa_checked_out_$BUILDROOT_VERSION
+
+do_if br_patch $BUILDROOT_DIR/.lisa_patched
+
+do_if br_apply_config $BUILDROOT_DIR/.lisa_applied_config
+
+make -C $BUILDROOT_DIR
+
+# Generate ssh-keys so that they persist across multiple runs
+ssh-keygen -A -f $BUILDROOT_DIR/output/target/
+# Enable root login on sshd
+sed -i 's/#PermitRootLogin.*/PermitRootLogin	yes/' $BUILDROOT_DIR/output/target/etc/ssh/sshd_config
```

Comment:
Is matching the "#" necessary (.* instead) ? If the conf file was changed to "PermitRootLogin no", not matching the # would still work

Reply:
Yes it is necessary otherwise we'll replace every occurenance in the file instead.

If the file was changed manually then we probably shouldn't be touching it. I don't think it's nice to override a user change behind their back - it should provide something that works out of the box, and if the user does some other  customisations on top it's better not to undo them.

Reply:
Agreed